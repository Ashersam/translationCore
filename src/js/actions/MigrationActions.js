// helpers
import {
  copySourceContentUpdaterManifest,
  getMissingResources,
  areResourcesNewer,
  removeOldThelps
} from '../helpers/ResourcesHelpers';

/**
 * Run migrations on the user tc resources folder. If it is determined the resources folder was
 * created before the source content updater then it will be deleted and a new resources folder
 * will be copied from tc to the users folder.
 */
export function migrateResourcesFolder() {
  return (() => {
    console.log("migrateResourcesFolder");
    if (areResourcesNewer()) {
      console.log("migrateResourcesFolder: copying newer resources");
      removeOldThelps();
      getMissingResources();
      copySourceContentUpdaterManifest();// Add source-content-updater-manifest.json
    } else {
      getMissingResources();
    }
  });
}

