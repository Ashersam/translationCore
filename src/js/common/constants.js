import path from "path-extra";
import ospath from "ospath";
import packagefile from '../../../package.json';

export const APP_VERSION = packagefile.version;
export const MIN_COMPATIBLE_VERSION = packagefile.minCompatibleVersion;

export const PROJECTS_PATH = path.join(ospath.home(), 'translationCore', 'projects');
export const USER_RESOURCES_PATH = path.join(ospath.home(), "translationCore", "resources");
export const IMPORTS_PATH = path.join(ospath.home(), 'translationCore', 'imports');
export const PROJECT_INDEX_FOLDER_PATH = path.join('.apps', 'translationCore', 'index');
export const TEMP_IMPORT_PATH = path.join(ospath.home(), 'translationCore', 'imports', 'temp');

export const PROJECT_DOT_APPS_PATH = path.join('.apps', 'translationCore');
export const PROJECT_CHECKDATA_DIRECTORY = path.join(PROJECT_DOT_APPS_PATH, 'checkData');
export const STATIC_RESOURCES_PATH = path.join(__dirname, "../../../tcResources");

export const TC_VERSION = "tc_version";
export const SOURCE_CONTENT_UPDATER_MANIFEST = "source-content-updater-manifest.json";

export const tc_EDIT_VERSION_KEY = "tc_edit_version"; // do not change this string as it will break compatibility with other versions
export const tc_MIN_COMPATIBLE_VERSION_KEY = "tc_min_compatible_version"; // do not change this string as it will break compatibility with other versions
export const tc_MIN_VERSION_ERROR = "TC_MIN_VERSION_ERROR";

export const testResourcesPath = path.join('__tests__', 'fixtures', 'resources');
export const toolCardCategories = {
  'kt': 'Key Terms',
  'names': 'Names',
  'other_terms': 'Other Terms',

  'culture': 'Culture',
  'figures': 'Figures of Speech',
  'numbers': 'Numbers',
  'discourse': 'Discourse',
  'grammar': 'Grammar',
  'other': 'Other'
};
