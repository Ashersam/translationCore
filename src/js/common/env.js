// methods for accessing environment variables
// this is needed because many process.env values are no longer defined on the client side (looks to be
// security related).  To get the environment variable on client side can now use remote.process.env.

var isRunningClientSide = !process.env.HOME && !!window; // if environment not defined and we have a window
console.log('utils: running Client Side: ' + isRunningClientSide);
var processEnv = isRunningClientSide ? window.require('electron').remote.process.env : process.env;


/**
 * get .env contents
 * @return {*}
 */
function getBuild() {
  return processEnv.BUILD;
}

/**
 * get remote env
 * @return {*}
 */
function getEnv() {
  return processEnv;
}

const env = {
  getBuild,
  getEnv
};

module.exports = env;
