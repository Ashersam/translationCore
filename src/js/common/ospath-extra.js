// this is a patched ospath to work on client side in windows.  The problem was that ospath
// made use of process.env values which are no longer defined on the client side (looks to be
// security related).  To get the environment variable can now use remote.process.env.
// TODO: electronite - investigate better solution
var path = require('path');
var os = require('os');
const { remote } = window.require('electron');

console.log('process.platform: ' + JSON.stringify(process.platform));

function data () {
  switch (this.__platform || process.platform) {
    case 'win32': return path.resolve(remote.process.env.APPDATA); // replaced process.env.APPDATA
    case 'darwin': return path.resolve(path.join(home.call(this), 'Library/Application Support/'));
    default: return process.env.XDG_CONFIG_HOME
      ? path.resolve(process.env.XDG_CONFIG_HOME)
      : path.resolve(path.join(home.call(this), '.config/'));
  }
}

function desktop () {
  return path.join(home.call(this), 'Desktop');
}

function home () {
  // io.js >= 2.3
  if ('homedir' in os) return os.homedir();

  switch (this.__platform || process.platform) {
    case 'win32': return path.resolve(remote.process.env.USERPROFILE); // replaced process.env.USERPROFILE
    default: return path.resolve(process.env.HOME)
  }
}

function tmp () {
  switch (this.__platform || process.platform) {
    case 'win32': return path.resolve(remote.process.env.TEMP); // replaced remote.process.env.TEMP
    default: return path.resolve('/tmp');
  }
}

var ospath = {
  __platform: process.platform,
  data: data,
  desktop: desktop,
  home: home,
  tmp: tmp
};

module.exports = ospath;
