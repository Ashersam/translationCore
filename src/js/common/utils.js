import { ipcRenderer } from 'electron';
const { remote } = window.require('electron');

let dotenv = null; // for caching values

export function delay(ms) {
  return new Promise((resolve) =>
    setTimeout(resolve, ms)
  );
}

/**
 * get .env contents
 * @return {*}
 */
export function getDotEnv() {
  if (!dotenv) {
    const env = ipcRenderer.sendSync('get-env');
    console.log('getDotEnv() - env: ' + JSON.stringify(env));
    dotenv = env;
  }
  return dotenv || {};
}

/**
 * get .env contents
 * @return {*}
 */
export function getBuild() {
  return getDotEnv().BUILD;
}

/**
 * get remote env
 * @return {*}
 */
export function getRemoteEnv() {
  return remote.process.env;
}
