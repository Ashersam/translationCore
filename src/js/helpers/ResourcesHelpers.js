
import fs from 'fs-extra';
import path from 'path-extra';
// constant declarations
const BIBLE_RESOURCES_PATH = path.join(path.homedir(), 'translationCore/resources/bibles');
const THELPS_RESOURCES_PATH = path.join(path.homedir(), 'translationCore/resources/translationHelps');
const STATIC_RESOURCES_BIBLES_PATH = './static/resources/bibles';
const STATIC_RESOURCES_THELPS_PATH = './static/resources/translationHelps';

/**
 * @description moves all bibles from the static folder to the local user translationCore folder.
 */
export function getBibleFromStaticPackage() {
  let bibleNames = fs.readdirSync(STATIC_RESOURCES_BIBLES_PATH);
  bibleNames.forEach((bibleName) => {
    let bibleSourcePath = path.join(STATIC_RESOURCES_BIBLES_PATH, bibleName);
    let bibleDestinationPath = path.join(BIBLE_RESOURCES_PATH, bibleName);
    if(!fs.existsSync(bibleDestinationPath)) {
      fs.copySync(bibleSourcePath, bibleDestinationPath);
    }
  });
}

/**
 * @description moves all translationHelps from the static folder to the resources folder in the translationCore folder.
 */
export function getTHelpsFromStaticPackage() {
  let tHelpsNames = fs.readdirSync(STATIC_RESOURCES_THELPS_PATH);
  tHelpsNames.forEach((tHelpName) => {
    let tHelpSourcePath = path.join(STATIC_RESOURCES_THELPS_PATH, tHelpName);
    let tHelpDestinationPath = path.join(THELPS_RESOURCES_PATH, tHelpName);
    if(!fs.existsSync(tHelpDestinationPath)) {
      fs.copySync(tHelpSourcePath, tHelpDestinationPath);
    }
  });
}