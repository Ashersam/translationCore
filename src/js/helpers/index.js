import {generateTimestamp} from './TimestampGenerator.js';

export {
  generateTimestamp
};
