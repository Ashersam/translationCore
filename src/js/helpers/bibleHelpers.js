/* eslint-disable no-console */
import Path from 'path-extra';
import fs from 'fs-extra';
import BooksOfBible from '../../../tC_resources/resources/books';
import {getLatestVersionInPath} from "./ResourcesHelpers";

/**
 *
 * @param {string} bookAbbr - The book abbreviation to convert
 */
export function convertToFullBookName(bookAbbr) {
  if (!bookAbbr) return;
  return BooksOfBible[bookAbbr.toString().toLowerCase()];
}

/**
 *
 * @param {string} projectBook - Book abbreviation
 */
export function isOldTestament(projectBook) {
  var passedBook = false;
  for (var book in BooksOfBible) {
    if (book == projectBook) passedBook = true;
    if (BooksOfBible[book] == 'Malachi' && passedBook) {
      return true;
    }
  }
  return false;
}

/**
 * This checks if a project is missing verses.
 * @param projectDir
 * @param bookId
 * @param resourceDir
 * @return {boolean}
 */
export const isProjectMissingVerses = (projectDir, bookId, resourceDir) => {
  try {
    let languageId = 'en';
    const resourcePath = Path.join(resourceDir, languageId, 'bibles', 'ulb');
    const versionPath = getLatestVersionInPath(resourcePath);
    const indexLocation = Path.join(versionPath, 'index.json');
    const expectedVerses = fs.readJSONSync(indexLocation);
    const actualVersesObject = {};
    const currentFolderChapters = fs.readdirSync(Path.join(projectDir, bookId));
    let chapterLength = 0;
    for (let currentChapterFile of currentFolderChapters) {
      let currentChapter = Path.parse(currentChapterFile).name;
      if (!parseInt(currentChapter)) continue;
      chapterLength++;
      let verseLength = 0;
      try {
        let currentChapterObject = fs.readJSONSync(
          Path.join(projectDir, bookId, currentChapterFile)
        );
        for (let verseIndex in currentChapterObject) {
          let verse = currentChapterObject[verseIndex];
          if (verse && verseIndex > 0) verseLength++;
        }
      } catch (e) {
        console.warn(e);
      }
      actualVersesObject[currentChapter] = verseLength;
    }
    actualVersesObject.chapters = chapterLength;
    let currentExpectedVerese = expectedVerses[bookId];
    return (
      JSON.stringify(currentExpectedVerese) !==
      JSON.stringify(actualVersesObject)
    );
  } catch (e) {
    console.warn(
      'ulb index file not found missing verse detection is invalid. Please delete ~/translationCore/resources folder'
    );
    return false;
  }
};
