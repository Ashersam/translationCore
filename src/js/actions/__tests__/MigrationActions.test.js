import path from 'path';
import fs from "fs-extra";
// helpers
import * as MigrationActions from "../MigrationActions";
// constants
import {APP_VERSION, STATIC_RESOURCES_PATH, TC_VERSION, USER_RESOURCES_PATH} from '../../common/constants';
import {getFoldersInResourceFolder} from "../../helpers/ResourcesHelpers";

// constants
const mockConsole = console;
jest.mock('../../helpers/ResourcesHelpers', () => ({
  ...require.requireActual('../../helpers/ResourcesHelpers'),
  extractZippedResourceContent: (resourceDestinationPath, isBible) => {
    mockConsole.log(`mock extractZippedResourceContent: resourceDestinationPath=${resourceDestinationPath} isBible=${isBible}`);
  }
}));

describe("migrate tCore resources", () => {
  beforeEach(() => {
    fs.__resetMockFS();
    // simulate static resources path
    fs.__loadFilesIntoMockFs(['resources'], path.join('__tests__', 'fixtures'), path.join(STATIC_RESOURCES_PATH, ".."));
    fs.moveSync(path.join(STATIC_RESOURCES_PATH, "../resources"), STATIC_RESOURCES_PATH);
    fs.__loadFilesIntoMockFs(['source-content-updater-manifest.json'], STATIC_RESOURCES_PATH, STATIC_RESOURCES_PATH);
  });

  it("test with no user resources", () => {
    // given
    const migrateResourcesFolder = MigrationActions.migrateResourcesFolder();

    // when
    migrateResourcesFolder();
    // then
    const folders = getResourceFolders();
    expect(folders).toMatchSnapshot();
  });

  it("test with t4t - should not delete", () => {
    // given
    fs.copySync(path.join(STATIC_RESOURCES_PATH, "en/bibles/ult"), path.join(USER_RESOURCES_PATH, "en/bibles/t4t"));
    const migrateResourcesFolder = MigrationActions.migrateResourcesFolder();

    // when
    migrateResourcesFolder();

    // then
    const folders = getResourceFolders();
    expect(folders).toMatchSnapshot();
  });

  it("test with t4t, old resources and up to date resources - should not delete anything", () => {
    // given
    fs.copySync(STATIC_RESOURCES_PATH, USER_RESOURCES_PATH);
    const manifestPath = path.join(USER_RESOURCES_PATH, 'source-content-updater-manifest.json');
    const manifest = fs.readJsonSync(manifestPath);
    manifest[TC_VERSION] = APP_VERSION; // add app version to resource
    fs.outputJsonSync(manifestPath, manifest);
    fs.copySync(path.join(STATIC_RESOURCES_PATH, "en/bibles/ult"), path.join(USER_RESOURCES_PATH, "en/bibles/t4t"));
    fs.copySync(path.join(STATIC_RESOURCES_PATH, "hi/translationHelps/translationWords"), path.join(USER_RESOURCES_PATH, "x-test/translationHelps/translationWords"));
    const migrateResourcesFolder = MigrationActions.migrateResourcesFolder();

    // when
    migrateResourcesFolder();

    // then
    const folders = getResourceFolders();
    expect(folders).toMatchSnapshot();
  });

  it("test with t4t, old resources, up to date resources and different app version - should remove old helps and keep old bibles", () => {
    // given
    fs.copySync(STATIC_RESOURCES_PATH, USER_RESOURCES_PATH);
    const manifestPath = path.join(USER_RESOURCES_PATH, 'source-content-updater-manifest.json');
    const manifest = fs.readJsonSync(manifestPath);
    manifest[TC_VERSION] = "x.x.x"; // add different version to resource
    fs.outputJsonSync(manifestPath, manifest);
    fs.copySync(path.join(STATIC_RESOURCES_PATH, "en/bibles/ult"), path.join(USER_RESOURCES_PATH, "en/bibles/t4t"));
    fs.copySync(path.join(STATIC_RESOURCES_PATH, "hi/translationHelps/translationWords"), path.join(USER_RESOURCES_PATH, "x-test/translationHelps/translationWords"));
    const migrateResourcesFolder = MigrationActions.migrateResourcesFolder();

    // when
    migrateResourcesFolder();

    // then
    const folders = getResourceFolders();
    expect(folders).toMatchSnapshot();
  });

  it("test with t4t, old resources and not up to date resources - should remove old helps and keep old bibles", () => {
    // given
    fs.copySync(STATIC_RESOURCES_PATH, USER_RESOURCES_PATH);
    const manifestPath = path.join(USER_RESOURCES_PATH, 'source-content-updater-manifest.json');
    const manifest = fs.readJsonSync(manifestPath);
    manifest[TC_VERSION] = APP_VERSION; // add app version to resource
    manifest.modified = "2014" + manifest.modified.substr(4);
    fs.outputJsonSync(manifestPath, manifest);
    fs.copySync(path.join(STATIC_RESOURCES_PATH, "en/bibles/ult"), path.join(USER_RESOURCES_PATH, "en/bibles/t4t"));
    fs.copySync(path.join(STATIC_RESOURCES_PATH, "hi/translationHelps/translationWords"), path.join(USER_RESOURCES_PATH, "x-test/translationHelps/translationWords"));
    const migrateResourcesFolder = MigrationActions.migrateResourcesFolder();

    // when
    migrateResourcesFolder();

    // then
    const folders = getResourceFolders();
    expect(folders).toMatchSnapshot();
  });

  it("test with t4t, old resources and up to date resources - should not delete anything", () => {
    // given
    fs.copySync(STATIC_RESOURCES_PATH, USER_RESOURCES_PATH);
    const manifestPath = path.join(USER_RESOURCES_PATH, 'source-content-updater-manifest.json');
    const manifest = fs.readJsonSync(manifestPath);
    manifest[TC_VERSION] = APP_VERSION; // add app version to resource
    manifest.modified = "9999" + manifest.modified.substr(4);
    fs.outputJsonSync(manifestPath, manifest);
    fs.copySync(path.join(STATIC_RESOURCES_PATH, "en/bibles/ult"), path.join(USER_RESOURCES_PATH, "en/bibles/t4t"));
    fs.copySync(path.join(STATIC_RESOURCES_PATH, "hi/translationHelps/translationWords"), path.join(USER_RESOURCES_PATH, "x-test/translationHelps/translationWords"));
    const migrateResourcesFolder = MigrationActions.migrateResourcesFolder();

    // when
    migrateResourcesFolder();

    // then
    const folders = getResourceFolders();
    expect(folders).toMatchSnapshot();
  });

  it("test with old resources - should delete", () => {
    // given
    fs.copySync(path.join(STATIC_RESOURCES_PATH, "hi/translationHelps/translationWords"), path.join(USER_RESOURCES_PATH, "x-test/translationHelps/translationWords"));
    const migrateResourcesFolder = MigrationActions.migrateResourcesFolder();

    // when
    migrateResourcesFolder();

    // then
    const folders = getResourceFolders();
    expect(folders).toMatchSnapshot();
  });
});

//
// Helpers
//

function toLinuxPath(filePath) {
  const newPath = filePath.split(path.sep).join(path.posix.sep);
  return newPath;
}

function getResourceFolders() {
  const paths = [];
  const languages = getFoldersInResourceFolder(USER_RESOURCES_PATH);
  for (let language of languages) {
    const resourceTypesPath = path.join(USER_RESOURCES_PATH, language);
    const resourceTypes = getFoldersInResourceFolder(resourceTypesPath);
    for (let resourceTYpe of resourceTypes) {
      const resourcesPath = path.join(resourceTypesPath, resourceTYpe);
      const resources = getFoldersInResourceFolder(resourcesPath);
      for (let resource of resources) {
        const resourcePath = toLinuxPath(path.join(resourcesPath, resource));
        paths.push(resourcePath);
      }
    }
  }
  return paths;
}
