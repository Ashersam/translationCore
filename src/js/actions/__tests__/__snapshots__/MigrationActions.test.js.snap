// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`migrate tCore resources test with no user resources 1`] = `
Array [
  "<HOME_DIR>/translationCore/resources/el-x-koine/bibles/ugnt",
  "<HOME_DIR>/translationCore/resources/el-x-koine/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/en/bibles/ult",
  "<HOME_DIR>/translationCore/resources/en/bibles/ust",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationAcademy",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationNotes",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/hbo/bibles/uhb",
  "<HOME_DIR>/translationCore/resources/hi/translationHelps/translationWords",
]
`;

exports[`migrate tCore resources test with old resources - should delete 1`] = `
Array [
  "<HOME_DIR>/translationCore/resources/el-x-koine/bibles/ugnt",
  "<HOME_DIR>/translationCore/resources/el-x-koine/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/en/bibles/ult",
  "<HOME_DIR>/translationCore/resources/en/bibles/ust",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationAcademy",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationNotes",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/hbo/bibles/uhb",
  "<HOME_DIR>/translationCore/resources/hi/translationHelps/translationWords",
]
`;

exports[`migrate tCore resources test with t4t - should not delete 1`] = `
Array [
  "<HOME_DIR>/translationCore/resources/en/bibles/t4t",
  "<HOME_DIR>/translationCore/resources/en/bibles/ult",
  "<HOME_DIR>/translationCore/resources/en/bibles/ust",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationAcademy",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationNotes",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/el-x-koine/bibles/ugnt",
  "<HOME_DIR>/translationCore/resources/el-x-koine/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/hbo/bibles/uhb",
  "<HOME_DIR>/translationCore/resources/hi/translationHelps/translationWords",
]
`;

exports[`migrate tCore resources test with t4t and up to date resources - should not delete 1`] = `
Array [
  "<HOME_DIR>/translationCore/resources/el-x-koine/bibles/ugnt",
  "<HOME_DIR>/translationCore/resources/el-x-koine/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/en/bibles/ult",
  "<HOME_DIR>/translationCore/resources/en/bibles/ust",
  "<HOME_DIR>/translationCore/resources/en/bibles/t4t",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationAcademy",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationNotes",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/hbo/bibles/uhb",
  "<HOME_DIR>/translationCore/resources/hi/translationHelps/translationWords",
]
`;

exports[`migrate tCore resources test with t4t, old resources and not up to date resources - should remove old helps and keep old bibles 1`] = `
Array [
  "<HOME_DIR>/translationCore/resources/el-x-koine/bibles/ugnt",
  "<HOME_DIR>/translationCore/resources/el-x-koine/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/en/bibles/ult",
  "<HOME_DIR>/translationCore/resources/en/bibles/ust",
  "<HOME_DIR>/translationCore/resources/en/bibles/t4t",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationAcademy",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationNotes",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/hbo/bibles/uhb",
  "<HOME_DIR>/translationCore/resources/hi/translationHelps/translationWords",
]
`;

exports[`migrate tCore resources test with t4t, old resources and up to date resources - should not delete anything 1`] = `
Array [
  "<HOME_DIR>/translationCore/resources/el-x-koine/bibles/ugnt",
  "<HOME_DIR>/translationCore/resources/el-x-koine/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/en/bibles/ult",
  "<HOME_DIR>/translationCore/resources/en/bibles/ust",
  "<HOME_DIR>/translationCore/resources/en/bibles/t4t",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationAcademy",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationNotes",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/hbo/bibles/uhb",
  "<HOME_DIR>/translationCore/resources/hi/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/x-test/translationHelps/translationWords",
]
`;

exports[`migrate tCore resources test with t4t, old resources and up to date resources - should not delete anything 2`] = `
Array [
  "<HOME_DIR>/translationCore/resources/el-x-koine/bibles/ugnt",
  "<HOME_DIR>/translationCore/resources/el-x-koine/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/en/bibles/ult",
  "<HOME_DIR>/translationCore/resources/en/bibles/ust",
  "<HOME_DIR>/translationCore/resources/en/bibles/t4t",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationAcademy",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationNotes",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/hbo/bibles/uhb",
  "<HOME_DIR>/translationCore/resources/hi/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/x-test/translationHelps/translationWords",
]
`;

exports[`migrate tCore resources test with t4t, old resources, up to date resources and different app version - should remove old helps and keep old bibles 1`] = `
Array [
  "<HOME_DIR>/translationCore/resources/el-x-koine/bibles/ugnt",
  "<HOME_DIR>/translationCore/resources/el-x-koine/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/en/bibles/ult",
  "<HOME_DIR>/translationCore/resources/en/bibles/ust",
  "<HOME_DIR>/translationCore/resources/en/bibles/t4t",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationAcademy",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationNotes",
  "<HOME_DIR>/translationCore/resources/en/translationHelps/translationWords",
  "<HOME_DIR>/translationCore/resources/hbo/bibles/uhb",
  "<HOME_DIR>/translationCore/resources/hi/translationHelps/translationWords",
]
`;
